drgn/helpers/linux/dmesg.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+dmesg-vmcore
+-------
+
+The ``drgn.helpers.linux.dmesg`` module provides helpers for reading the
+Linux kernel log.
+"""
+
+from drgn import Program, Object, cast, sizeof
+
+__all__ = ("get_dmesg",)
+
+DESC_SV_BITS = 0
+DESC_FLAGS_SHIFT = 0
+DESC_FLAGS_MASK = 0
+DESC_ID_MASK = 0
+
+def id_incr(counter):
+    return (counter + 1) & DESC_ID_MASK
+
+def record_committed(current_id, state_var):
+    state_desc_id = state_var & DESC_ID_MASK
+    state = 3 & (state_var >> DESC_FLAGS_SHIFT)
+    return (current_id == state_desc_id) and (state == 0x1 or state == 0x2)
```

Comment:
Instead of hardcoding 0x1 and 0x2, you can get the states as `prog['desc_committed']` and `prog['desc_finalized']`.